v5.0.0-alpha.32 (#25997)
mui-org/material-ui: CHANGELOG.md

### [Versions](https://material-ui.com/versions/)

## 5.0.0-alpha.32

<!-- generated comparing v5.0.0-alpha.31..next -->

_Apr 27, 2021_

Big thanks to the 15 contributors who made this release possible. Here are some highlights ✨:

- 👩‍🎤 We finished the migration of all components in `@material-ui/core`. We will focus on the components in `@material-ui/lab` next.

> - 💥 Resume work on the breaking changes, aim for v5.0.0-beta.0 in the next coming months.

Reviewer: ```suggestion
- 💥 Make progress with the breaking changes. We have done 38 out of 41 breaking changes that can be deprecated. We have done 21 out of the 39 that can't have deprecations. Once done, we will focus on updating the component for better following material design, and to improve the aesthetic.
```

> - 📚 Add TypeScript support for adding custom `size` and `color` values in all components.

Reviewer: prop-types were also changed, it's not only TypeScript.

```suggestion
- 💄 Support extending the theme for custom color and size values in all components.
```

### `@material-ui/styled-engine@5.0.0-alpha.32`

- <!-- 02 --> [styled-engine] Skip variants resolver for non root slots by default (#25865) @mnajdova

### Core

- <!-- 54 --> [core] Fix wrong imports to '@material-ui/styles' (#25984) @mnajdova
- <!-- 52 --> [core] Ensure props spreading works as expected (#25939) @oliviertassinari
- <!-- 47 --> [core] Batch small changes (#25968) @oliviertassinari
- <!-- 45 --> [core] Fix various too wide `classes` types (#25917) @eps1lon
- <!-- 35 --> [core] Enable trailing comma in TypeScript files (#25931) @eps1lon
- <!-- 33 --> [core] Remove @typescript-to-proptypes-generate handlers (#25909) @eps1lon
- <!-- 18 --> [core] Update slots components to enable flatten specificity for overrides (#25853, #25864, #25881, #25884, #25887, #25904, #25892) @mnajdova
- <!-- 05 --> [Drawer] Fix classes forwarded to DOM node for docked drawer (#25870) @mnajdova
- <!-- 12 --> [system] Add missing `main` entry for styleFunctionSx (#25885) @eps1lon
- <!-- 27 --> [test] Add current behavior of inverleaving elements on mousedown (#25903) @eps1lon
- <!-- 20 --> [test] Add test validator to improve DX (#25854) @siriwatknp

> ### Docs

Reviewer: So far, we have made docs come before core before more important

Author: Moved the Docs before the Core

### [Versions](https://material-ui.com/versions/)

## 5.0.0-alpha.32

<!-- generated comparing v5.0.0-alpha.31..next -->

_Apr 27, 2021_

Big thanks to the 15 contributors who made this release possible. Here are some highlights ✨:

- 👩‍🎤 We have completed the migration to emotion of all components in `@material-ui/core`. We will focus on the components in `@material-ui/lab` next.

> - 💥 Make progress with the breaking changes. We have done 38 out of 41 breaking changes that can be deprecated. We have done 21 out of the 39 that can't have deprecations. Once done, we will focus on updating the component for better following material design, and to improve the aesthetic.

Reviewer: Make to highlight that we don't plan more

```suggestion
- 💥 Make progress with the breaking changes plan. We have done 38 out of 41 breaking changes that can be deprecated. We have done 21 out of the 39 that can't have deprecations. Once done, we will focus on updating the component for better following material design, and to improve the aesthetic.
```

- 📚 Add TypeScript support for adding custom `size` and `color` values in all components.
- And many more 🐛 bug fixes and 📚 improvements.

### `@material-ui/core@5.0.0-alpha.32`

#### Breaking changes

- <!-- 46 -->  [Table] Rename padding="default" to padding="normal" (#25924) @m4theushw

  ```diff
  -<Table padding="default" />
  -<TableCell padding="default" />
  +<Table padding="normal" />
  +<TableCell padding="normal" />
  ```

- <!-- 29 --> [Button] Rename `pending` prop to `loading` in LoadingButton (#25874) @m4theushw

  ```diff
  -<LoadingButton pending pendingIndicator="Pending..." pendingPosition="end" />
  +<LoadingButton loading loadingIndicator="Pending..." loadingPosition="end" />
  ```

- <!-- 25 --> [ButtonBase] Remove buttonRef prop (#25896) @m4theushw

>   ```diff
>   -<ButtonBase buttonRef={ref} />
>   +<ButtonBase ref={ref} />
> 
>   -<Button buttonRef={ref} />
>   +<Button ref={ref} />
>   ```

Reviewer: The blank line doesn't show once inside the release page of GitHub

````suggestion
  ```diff
  -<ButtonBase buttonRef={ref} />
  +<ButtonBase ref={ref} />
  ```

  ```diff
  -<Button buttonRef={ref} />
  +<Button ref={ref} />
  ```
````

- <!-- 40 --> [Timeline] Remove use of nth-child in favor of nth-of-type (#25915) @wellwellmissesanderson
- <!-- 06 --> [Timeline] Migrate Timeline to emotion (#25838) @siriwatknp
- <!-- 55 --> [TreeView] Migrate TreeItem to emotion (#25835) @siriwatknp

### `@material-ui/styled-engine@5.0.0-alpha.32`

- <!-- 02 --> [styled-engine] Skip variants resolver for non root slots by default (#25865) @mnajdova

### `@material-ui/types@6.0.0`

This package is just re-released since version 5.1.7 had a breaking change.

### Core

- <!-- 54 --> [core] Fix wrong imports to '@material-ui/styles' (#25984) @mnajdova
- <!-- 52 --> [core] Ensure props spreading works as expected (#25939) @oliviertassinari
- <!-- 47 --> [core] Batch small changes (#25968) @oliviertassinari
- <!-- 45 --> [core] Fix various too wide `classes` types (#25917) @eps1lon
- <!-- 35 --> [core] Enable trailing comma in TypeScript files (#25931) @eps1lon
- <!-- 33 --> [core] Remove @typescript-to-proptypes-generate handlers (#25909) @eps1lon
- <!-- 18 --> [core] Update slots components to enable flatten specificity for overrides (#25853, #25864, #25881, #25884, #25887, #25904, #25892) @mnajdova

> - <!-- 05 --> [Drawer] Fix classes forwarded to DOM node for docked drawer (#25870) @mnajdova

Reviewer: Wrong location? Should be under the core package?

> - <!-- 12 --> [system] Add missing `main` entry for styleFunctionSx (#25885) @eps1lon

Reviewer: Wrong location? Should be under the system package?

- <!-- 27 --> [test] Add current behavior of inverleaving elements on mousedown (#25903) @eps1lon
- <!-- 20 --> [test] Add test validator to improve DX (#25854) @siriwatknp

### Docs

- <!-- 28 --> [Autocomplete] Fix tagSize class typo (#25908) @JanMisker
- <!-- 51 --> [DataGrid] Update docs sections (#25980) @dtassone
- <!-- 38 --> [docs] Batch small fixes (#25807) @m4theushw
- <!-- 37 --> [docs] Grammar correction in autocomplete API (#25910) @gruber76
- <!-- 36 --> [docs] Update minimum required TypeScript version (#25930) @eps1lon
- <!-- 31 --> [docs] Fix Typography api docs for `paragraph` prop (#25929) @DanailH
- <!-- 17 --> [docs] Fix Slider's classes wrong description (#25907) @mnajdova
- <!-- 15 --> [docs] Require documentation of demos (#25811) @eps1lon
- <!-- 13 --> [docs] Explicitly list demos of unstyled components (#25900) @eps1lon
- <!-- 09 --> [docs] Fix global styles leaking on different pages (#25855) @mnajdova
- <!-- 04 --> [docs] Expose heading links in a11y tree (#25861) @eps1lon

> - <!-- 16 --> [l10n] Add Bengali (bnBD) locale (#25841) @Knoxo

Reviewer: Wrong location? Should be under the core package?

- <!-- 13 --> [docs] Explicitly list demos of unstyled components (#25900) @eps1lon
- <!-- 09 --> [docs] Fix global styles leaking on different pages (#25855) @mnajdova
- <!-- 04 --> [docs] Expose heading links in a11y tree (#25861) @eps1lon
- <!-- 56 --> [Table] Improve description of TablePagination.rowsPerPageOptions (#25982) @kevinlul

### Core

- <!-- 54 --> [core] Fix wrong imports to '@material-ui/styles' (#25984) @mnajdova
- <!-- 52 --> [core] Ensure props spreading works as expected (#25939) @oliviertassinari
- <!-- 47 --> [core] Batch small changes (#25968) @oliviertassinari

> - <!-- 45 --> [core] Fix various too wide `classes` types (#25917) @eps1lon

Reviewer 2: part of `@material-ui/core`. You should include the affected components (listed in the PR description).

- <!-- 35 --> [core] Enable trailing comma in TypeScript files (#25931) @eps1lon
- <!-- 33 --> [core] Remove @typescript-to-proptypes-generate handlers (#25909) @eps1lon
- <!-- 18 --> [core] Update slots components to enable flatten specificity for overrides (#25853, #25864, #25881, #25884, #25887, #25904, #25892) @mnajdova

> - <!-- 16 --> [l10n] Add Bengali (bnBD) locale (#25841) @Knoxo

Reviewer 2: part of `@material-ui/core`
